Formula/retroshare-service.rb

```diff
@@ -0,0 +1,29 @@
+class RetroshareService < Formula
+  desc "RetroShare friend to friend communication headless service"
+  homepage "https://retroshare.cc/"
+  url "https://nuvula.diveni.re/s/zBHagxfpGFeDxRA/download/RetroShare-v0.6.6-489-g000d01994-homebrew.tar.gz"
+  sha256 "4670e605ed165907cb059335b0b034cbfb5b7638525209c8339b0e4b1d863f0e"
+  license "AGPL-3.0-only"
+  head "https://github.com/RetroShare/RetroShare.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "doxygen" => :build
+  depends_on "python@3.9" => :build
+  depends_on "rapidjson" => :build
+  depends_on "miniupnpc"
+  depends_on "openssl@1.1"
+  depends_on "sqlcipher"
+  depends_on "xapian"
+
+  def install
+    system "cmake", "-S", "retroshare-service", "-B", "builddir",
+                    "-DRS_FORUM_DEEP_INDEX=ON",
+                    *std_cmake_args
+    system "cmake", "--build", "builddir"
+    system "cmake", "--install", "builddir"
+  end
+
+  test do
+    puts shell_output("find #{prefix}")
```

Comment:
We need a test that exercises the some of the functionality of the app.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks @SMillerDev should be ready for review and hopefully merging now

Reply:
`--help` isn't really testing anything either. It should really check if the app works.

Reply:
In all other more realistic usages retroshare-service need to keep running while other commands `curl` need to query it's http API, how to do that in the formula?

Reply:
You can check httpd for an example of that.

Reply:
now the tests are serious, thanks for suggestion